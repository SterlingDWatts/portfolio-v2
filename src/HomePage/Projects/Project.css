.Project__header {
    margin: 32px 0 16px;
    text-align: center;
    font-size: 32px;
}

.Project__pic {
    width: 100%;
    height: calc((100vw - 32px) * 1.883);
    background-size: cover;
    background-position: top;
    box-shadow: 6px 6px 5px 0 rgba(0, 0, 0, 0.5), 0px 0px 10px 4px rgba(0,0,0,0.25);
    position: relative;
}

.Project__about {
    position: absolute;
    bottom: 8px;
    left: 8px;
    right: 8px;
    padding: 8px;
    background-color: #ffff6b;
    border-radius: 5px;
    box-shadow: 4px 4px 5px 0 #8383368a, 0px 0px 10px 4px rgba(0,0,0,0.25);
}

.Project__more_info {
    float: right;
}

.Quizz_pic {
    background-image: url(https://sterlingdwatts.github.io/portfolio/images/quaff-quiz-mobile.png);
}

.Hub_pic {
    background-image: url(https://sterlingdwatts.github.io/portfolio/images/game-hub-mobile-landing-page.png);
    height: calc((100vw - 32px) * 1.628);
}

.Project a:link, .Project a:visited {
    color: #00685b;
}

@media screen and (min-width: 500px) {
    .Project__pic {
        width: 80vw;
        margin: 0 auto;
    }

    .Quizz_pic {
        background-image: url(https://sterlingdwatts.github.io/portfolio/images/quaff-quiz-desktop.png);
        height: calc(80vw * 0.666)
    }

    .Hub_pic {
        background-image: url(https://sterlingdwatts.github.io/portfolio/images/game-hub-desktop-landing-page.png);
        height: calc(80vw * 0.666);
    }
}

@media screen and (min-width: 900px) {
    .Project__pic, .Quizz_pic, .Hub_pic {
        width: 680px;
        height: 453.33px;
    }
}