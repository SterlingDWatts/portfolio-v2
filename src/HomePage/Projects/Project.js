import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Project.css';

class Project extends Component {

    state = {
        showMore: false
    }

    handleButtonClick = () => {
        this.setState({
            showMore: !this.state.showMore
        })
    }

    render() {
        return (
            <div className='Project'>
                <header className='Project__header'>
                    <h3>{this.props.name}</h3>
                </header>
                <div className={'Project__pic ' + this.props.name + '_pic'}>
                    <div className='Project__about'>
                        <h5>
                            {this.props.name}
                            {' | '}
                            <a href={this.props.site} target='_blank'>Live App</a>
                            {' | '}
                            <a href={this.props.repo} target='_blank'>Repo</a>
                        </h5>
                        <p>{this.props.blurb}</p>
                        {this.state.showMore && 
                            <p>{this.props.moreInfo}</p>
                        }
                        <button 
                            className='Project__more_info'
                            onClick={this.handleButtonClick}
                        >
                            more info
                        </button>
                        {this.state.showMore && 
                            <p>{this.props.tech}</p>
                        }
                    </div>
                </div>
                
            </div>
        );
    }

}

Project.propTypes = {
    name: PropTypes.string.isRequired,
    pic: PropTypes.string.isRequired,
    site: PropTypes.string.isRequired,
    repo: PropTypes.string.isRequired,
    blurb: PropTypes.string.isRequired,
    moreInfo: PropTypes.string.isRequired,
    tech: PropTypes.string.isRequired,
}

export default Project;
