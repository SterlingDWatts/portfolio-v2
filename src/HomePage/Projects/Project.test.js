import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Project from './Project';

describe(`Project Component`, () => {

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Project />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<Project />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

})
