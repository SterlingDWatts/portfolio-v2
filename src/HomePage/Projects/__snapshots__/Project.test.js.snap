// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Component renders the UI as expected 1`] = `
<div
  className="Project"
>
  <header
    className="Project__header"
  >
    <h3 />
  </header>
  <div
    className="Project__pic undefined_pic"
  >
    <div
      className="Project__about"
    >
      <h5>
         | 
        <a
          target="_blank"
        >
          Live App
        </a>
         | 
        <a
          target="_blank"
        >
          Repo
        </a>
      </h5>
      <p />
      <button
        className="Project__more_info"
        onClick={[Function]}
      >
        more info
      </button>
    </div>
  </div>
</div>
`;
